### ISSUE TITLE
NanoPi R5S now supported in OpenWRT

### ISSUE THREAD

**Issue by judahrand** [anaelorlinski/OpenWrt-NanoPi-R5S-Builds#11]
It looks like support for NanoPi R5S finally got merged and SNAPSHOT builds are now available. Until the official builds are known to be stable it certainly makes sense to maintain this repo but once we're sure the official builds are stable does maintaining the repo make sense?

https://firmware-selector.openwrt.org/?version=SNAPSHOT&target=rockchip%2Farmv8&id=friendlyarm_nanopi-r5s

**Comment by anaelorlinski** [comment]
Yes I know. My decision to keep or kill the builds will depend on the support of some features by the default ones.
Right now my builds use the Realtek driver for network instead of mainline one, and also support HDMI console with USB keyboard. I think this is not supported or will never be in mainline OpenWrt.
I try to keep the set of patches minimal but also to provide a correct set of packages out of the box.

**Comment by judahrand** [comment]
That's really useful context! I wonder if the the differences between the official builds vs these build would be useful in the README?

**Comment by ChrSch84** [comment]
Could i ask why is there a different between the Realtek driver and the mainline driver?

And maybe you know why HDMI and USB is not supportet at the Mainline Firmware? Thank you.

**Comment by anaelorlinski** [comment]
The realtek driver is provided to support features of the network chips when they are released, usually enabling new features immediately (such as 2.5G). The mainline kernel driver is updated with new features too but usually takes longer time for them to become available. The gap between the 2 drivers is reduced over time and also performance becomes similar. I guess the mainline driver is now fully usable with R5S.

Regarding HDMI I think it is by choice because most of OpenWrt devices don't have HDMI and are used as router so it does not make sense to add extra patches to support it until the support is provided by the mainline kernel. I think USB is supported though.

**Comment by ChrSch84** [comment]
Thanks for the info.

At the Moment i use my R5S with Nightly Builds of Openwrt. Network reaches 2,3gbit to a 10Gbit Backend.

But i miss the USB and HDMI Funktion. Do you know how to add this? Or is it too complicated.

Thanks

PS: Maybe make a other Thread.
